### PR TITLE
scheduler: remove FrameworkFactory.

### PR DESCRIPTION
**Pull request by tangwz**
**What type of PR is this?**
/kind cleanup
/sig scheduling

**What this PR does / why we need it**:
Removing `FrameworkFactory` and calling `runtime.NewFramework` directly after `Profile` is removed.
Related PR: https://github.com/kubernetes/kubernetes/pull/95687

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
@alculquicondor

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/ok-to-test
/triage accepted
/assign @hex108

@hex108 would you mind giving a first pass?

File: pkg/scheduler/factory.go [hunk 1]
```diff
@@ -149,8 +131,19 @@ func (c *Configurator) create() (*Scheduler, error) {
 
 	// The nominator will be passed all the way to framework instantiation.
 	nominator := internalqueue.NewPodNominator()
-	profiles, err := profile.NewMap(c.profiles, c.buildFramework, c.recorderFactory,
-		frameworkruntime.WithPodNominator(nominator))
+	for _, p := range c.profiles {
+		if c.frameworkCapturer != nil {
```
Comment (hex108) [inline pkg/scheduler/factory.go:135]: Would it be better to put this logic after this check https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/profile/profile.go#L63 ?

Comment (alculquicondor) [reply pkg/scheduler/factory.go:135]: That might be ideal, but I prefer we don't pass yet another parameter to `NewMap`. Maybe we could just do this loop after calling `NewMap`.

Comment (tangwz) [reply pkg/scheduler/factory.go:135]: I think it's depending on struct `c` and `c.frameworkCapturer()` is a private function.

Comment (tangwz) [reply pkg/scheduler/factory.go:135]: > Would it be better to put this logic after this check https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/profile/profile.go#L63 ?

I think it's depending on struct `c` and `c.frameworkCapturer()` is a private function.

File: pkg/scheduler/factory.go [hunk 2]
```diff
@@ -149,8 +131,19 @@ func (c *Configurator) create() (*Scheduler, error) {
...
+		}
+	}
+	opts := []frameworkruntime.Option{
+		frameworkruntime.WithClientSet(c.client),
+		frameworkruntime.WithInformerFactory(c.informerFactory),
+		frameworkruntime.WithSnapshotSharedLister(c.nodeInfoSnapshot),
+		frameworkruntime.WithRunAllFilters(c.alwaysCheckAllPredicates),
+		frameworkruntime.WithPodNominator(nominator),
+	}
+	profiles, err := profile.NewMap(c.profiles, c.registry, c.recorderFactory, opts...)
```
Comment (alculquicondor) [inline pkg/scheduler/factory.go:146]: nit: inline the opts. You can use multiple lines.

```
NewMap(c.profiles, c.registry, c.recorderFactory,
  frameworkruntime.WithClientSet(...),
  ...
```

**Review by alculquicondor — COMMENTED** [review]
/approve

Please squash

**Comment by tangwz** [comment]
@alculquicondor PTAL

**Pull request merged**